provider/server/exec.go

```diff
@@ -5,11 +5,165 @@ package server
 
 import (
 	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"os/exec"
+	"strings"
+	"sync"
+	"time"
 
+	"github.com/creack/pty"
+	"github.com/digitalocean/go-libvirt"
+	"github.com/go-logr/logr"
 	iri "github.com/ironcore-dev/ironcore/iri/apis/machine/v1alpha1"
+	remotecommandserver "github.com/ironcore-dev/ironcore/poollet/machinepoollet/iri/streaming/remotecommand"
+	"github.com/ironcore-dev/libvirt-provider/pkg/api"
+	libvirtutils "github.com/ironcore-dev/libvirt-provider/pkg/libvirt/utils"
+	"github.com/moby/term"
+	"k8s.io/client-go/tools/remotecommand"
 )
 
+const (
+	StreamCreationTimeout = 30 * time.Second
+	StreamIdleTimeout     = 2 * time.Minute
+)
+
+type executorExec struct {
+	Libvirt         *libvirt.Libvirt
+	ExecRequest     *iri.ExecRequest
+	VirshExecutable string
+	Machine         *api.Machine
+}
+
 func (s *Server) Exec(ctx context.Context, req *iri.ExecRequest) (*iri.ExecResponse, error) {
+	machineID := req.MachineId
+	log := s.loggerFrom(ctx, "MachineID", machineID)
+
+	log.V(1).Info("Inserting request into cache")
+	token, err := s.execRequestCache.Insert(req)
+	if err != nil {
+		return nil, err
+	}
+
+	log.V(1).Info("Returning url with token")
+	return &iri.ExecResponse{
+		Url: s.buildURL("exec", token),
+	}, nil
+}
```

Comment:
Can you please return a `not found error` if the machine is not in the store?

Reply:
Sure, I added a check to verify the store, before returning the token.